config.json

```
@@ -2464,6 +2464,26 @@
           "multiple-clause-matching"
         ],
         "difficulty": 5
+      },
+      {
+        "slug": "affine-cipher",
+        "name": "Affine Cipher",
+        "uuid": "18d12bac-ead0-456c-bcef-ccb2fd06fe72",
+        "prerequisites": [
+          "pattern-matching",
+          "atoms",
+          "pipe-operator",
+          "enum",
+          "integers",
+          "if",
+          "strings",
+          "tuples"
```

The key is a map, so we also need to require maps.

For the code that actually changes the letter to a different one, we need either charlists (your approach) or list comprehensions (alternative approach)
```elixir
      message =
        encrypted
        |> String.replace(@ignored, "")
        |> to_charlist()
        |> Enum.map(fn
          digit when ?0 <= digit and digit <= ?9 -> digit
          char -> Integer.mod(mmi * (char - ?a - b), @alphabet_size) + ?a
        end)
        |> to_string
```

```elixir
      message = String.replace(encrypted, @ignored, "")

      message =
        for << char <- message >>, into: <<>> do
          case char do
            char when ?0 <= char and char <= ?9 -> <<char>>
            char -> <<Integer.mod(mmi * (char - ?a - b), @alphabet_size) + ?a>>
          end
        end
```

```suggestion
          "tuples",
          "maps",
          "list-comprehensions",
          "charlists"
```